source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,276 @@
+   import org.bson.Document;
+
+   import java.util.Arrays;
+
+Connect to a MongoDB Deployment
++++++++++++++++++++++++++++++++
+
+Connect to a MongoDB deployment and declare and define ``MongoDatabase`` and ``MongoCollection`` instances.
+
+For example, include the following code to connect to a standalone MongoDB deployment running on localhost on port
+``27017`` and define ``database`` to refer to the ``test`` database and ``collection`` to refer to the ``restaurants``
+collection.
+
+.. code-block:: java
+
+   public class AggTour {
+
+       public static void main(String[] args) {
+           MongoClient mongoClient = MongoClients.create();
+           MongoDatabase database = mongoClient.getDatabase("test");
+           MongoCollection<Document> collection = database.getCollection("restaurants");
+
+           // aggregation here
+       }
+   }
+
+.. tip::
+
+   For information on connecting to MongoDB, see the :ref:`Connection Guide <mongoclient>`
+
+Basic Aggregation Example
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To perform an aggregation, pass a list of aggregation stages to the `MongoCollection.aggregate() <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/MongoCollection.html#aggregate(com.mongodb.client.ClientSession,java.util.List,java.lang.Class)>`__
+method.
+
+The Java driver provides the `Aggregates <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html>`__
+helper class that
+contains builders for aggregation stages.
+
+In the following example, the aggregation pipeline:
+
+- Uses a :manual:`$match <reference/operator/aggregation/match/>` stage to filter for documents whose
+  ``categories`` array field contains the element ``Bakery``. The example uses
+  `Aggregates.match <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#match(org.bson.conversions.Bson)>`__
+  to build the ``$match`` stage.
+
+- Uses  a :manual:`$group <reference/operator/aggregation/group/>` stage to group the matching documents by the ``stars``
+  field, accumulating a count of documents for each distinct value of ``stars``.
+
+.. note::
+
+   The example uses `Aggregates.group <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#group(org.bson.conversions.Bson)>`__
+   to build the ``$group`` stage and `Accumulators.sum <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators#sum(java.lang.String,TExpression)>`__
+   to build the accumulator expression. The :manual:`accumulator expressions <reference/operator/aggregation/#accumulators-group>`
+   used within the ``$group`` stage are built with the `Accumulators <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators.html>`__
+   helper class.
```

Comment:
I think we should use active voice to clearly identify the who and how. E.g. "You can build the accumulator expressions used within the $group stage using the Accumulators helper class. "

Reply:
I've removed most of this entirely, since it's getting too deep in the weeds and there are going to be other pages that cover these in more detail.